### PR TITLE
[#15056] Can't see display/ens/name when mentioning in community

### LINKED ISSUE #15056: Can't see display/ens/name when mentioning in community
When mentioning someone, I can only see 3 words names, we should display (and search on), the contact names.

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #15056

contacts partially has been refactored in this PR

now we have `:first-name` and `:second-name` fields in contact map

new component was introduced here `quo2.components.list-items.user-list`

QA pls take a look on places where we show usernames

### CODE REVIEW

**Review by ibrkhalil — COMMENTED** [review]
lgtm

**Review by ibrkhalil — APPROVED** [review]
lgtm

File: src/status_im2/subs/communities.cljs [hunk 1]
```diff
@@ -50,6 +50,13 @@
  (fn [communities [_ id]]
    (get-in communities [id :members])))
 
+(re-frame/reg-sub
+ :communities/current-community-members
+ :<- [:chats/current-chat]
```
Comment (Parveshdhull) [inline src/status_im2/subs/communities.cljs:55]: We have `:chats/current-chat` at several places, but it looks weird to me.
It's hard to follow its life cycle, for me it is better to pass an id instead of relying on some subscription. wdyt?
(probably ok in this case, as parent already using this subscription)

Comment (flexsurfer) [reply src/status_im2/subs/communities.cljs:55]: im not sure i understand, what's the problem, could you elaborate ?

Comment (Parveshdhull) [reply src/status_im2/subs/communities.cljs:55]: > Explicit is better than implicit


There were several scenarios where we didn't pass values explicitly, and that leads to a bug
1. **Pinned message bug** - Even though we have access to chat-id, we were relying on `:chats/current-chat`, even when the current chat was closed, which caused a bug. [Code](https://github.com/status-im/status-mobile/pull/14885/files#diff-bdf749d4861990bf570c189355c0808d58a17f1bc536124fe182746d87007781) (Line 33)

3. **Famous Unknown community bug** - Instead of passing view-id explicitly (even though we were accessing screen-param in communities screen), we assumed `:view-id` already has `:community` as its value, as we are navigating. Due to a weird scenario, this `view-id` was `:home-stack` and lead to a bug. [Code](https://github.com/status-im/status-mobile/pull/14410/files#diff-1abdb083e4a14a5552793ebe14ced407956e46df7e4a0eb86f25e0f7291eaa87) (Line 248)

I just want to raise the point, we should try to pass values explicitly, especially in the cases we have access to that (like above 2), instead of relying on some third subscription.

Comment (flexsurfer) [reply src/status_im2/subs/communities.cljs:55]: i mean, if you want a value for current chat it must be a current chat overwise there is a bug , so i still don't see the issue, if we pass chat-id but we want a current chat, and chat id is different from current chat something wrong, passing id is not always convenient, like, in this example, it was implemented incorrectly, so having current-chat subscription is convenient and more idiomatic i  would say

**Review by Parveshdhull — APPROVED** [review]

**Comment by churik** [comment]
The reason for failed e2e:
### ISSUE 1: no name is shown in profile

**Steps:**
1) mobile: create multiaccount
2) tap on profile tab


**Expected result:**
can see 3-random name

**Actual result:**
IMAGE 2023-02-22 13:08:19

**OS:** IOS, Android

File: src/status_im/data_store/contacts.cljs [hunk 2]
```diff
@@ -1,26 +1,63 @@
 (ns status-im.data-store.contacts
   (:require [clojure.set :as set]
             [utils.re-frame :as rf]
-            [taoensso.timbre :as log]))
+            [taoensso.timbre :as log]
+            [clojure.string :as string]
+            [status-im2.constants :as constants]
+            [status-im.utils.gfycat.core :as gfycat]))
+
+(defn assoc-added
```
Comment (cammellos) [inline src/status_im/data_store/contacts.cljs]: I think this is already there, you can use `added` property from status-go

File: src/status_im/data_store/contacts.cljs [hunk 3]
```diff
@@ -1,26 +1,63 @@
...
+       {:id                  :public-key
+        :compressedKey       :compressed-key
+        :ensVerified         :ens-verified
+        :hasAddedUs          :has-added-us
+        :contactRequestState :contact-request-state
+        :lastUpdated         :last-updated
+        :localNickname       :nickname
+        :displayName         :display-name
+        :blocked             :blocked?})
+      assoc-mutual
```
Comment (cammellos) [inline src/status_im/data_store/contacts.cljs]: All of these can be done in status-go, better, I can do if you'd like

File: src/status_im/multiaccounts/core.cljs [hunk 4]
```diff
@@ -12,84 +11,35 @@
...
-        ens-name     (or preferred-name
-                         display-name
-                         name)]
-    ;; Preferred name is our own otherwise we make sure it's verified
-    (if (or preferred-name (and ens-verified name))
-      (let [username (stateofus/username ens-name)]
-        (or username ens-name))
-      (or display-name alias (gfycat/generate-gfy public-key)))))
+  "Legacy"
+  [{:keys [first-name]}]
```
Comment (cammellos) [inline src/status_im/multiaccounts/core.cljs]: (def displayed-name :first-name) ;)

Comment (flexsurfer) [reply src/status_im/multiaccounts/core.cljs]: oh wow, took a minute to understand :)

Comment (cammellos) [reply src/status_im/multiaccounts/core.cljs]: it's not exactly the same, since the arity of the function is different, and sometimes that matters, for example:

```
(def displayed-name-1 :first-name)

(def displayed-name-2 [{:keys [first-name]}] first-name)
```
will give different result in cases like:

```
(displayed-name-1 {} "b") -> returns `b`
(displayed-name-2 {} "b") -> returns `nil`
```

and that's why we don't use it in `reg-sub`, where you might want to do

```
(re-frame/reg-sub
 :activity-center-state
 :field-on-state)
```

which looks neat, but you get the event name instead if the `activity-center-state` does not have the key, fun fact :)

**Comment by flexsurfer** [comment]
thanks @churik fixed

**Comment by pavloburykh** [comment]
Thanx for the PR @flexsurfer 
Please, take a look at the following issues

### ISSUE 2 1-1 chat is always started with the first user in contact list regardless which user has been selected on New chat screen

This bug has already been recently fixed by @Parveshdhull https://github.com/status-im/status-mobile/pull/15124 and is not reproducible in develop. 

Current PR should include this fix but I guess it is broken again due to some changes

**Steps:** 

1. Create/recover user who has at least 2 contacts 
2. Tap Chats tab
3. Tap + (plus) button => New chat
4. Select any user from the contact list except first one
5. See which user is in fact selected (observe text in appeared "Chat with ...." button
6. Tap "Chat with ...."
7. See with which user the chat has been started

**Expected result:** chat should start with exact user that has been selected from contact list. 

**Actual result:** 1-1 chat is always started with first user from contact list. 

[Status-debug-logs (5).zip](https://github.com/status-im/status-mobile/files/10812672/Status-debug-logs.5.zip)

https://user-images.githubusercontent.com/97245802/220885111-fa4b4503-723c-4553-88e3-a9df6b7e0dbb.mp4

**Comment by pavloburykh** [comment]
### ISSUE 3 "null is not an object (evaluating 'I.substring') error during and after ENS setup

**Steps:** 

1. Open ENS setup settings
2. Add ENS for current user
3. Observe an error after adding ENS
4. Try to re-enter ENS setup screen
5. Observe the same error

**Expected result:** no error

**Actual result:** "null is not an object (evaluating 'I.substring') error occurs at the moment when user adds ENS. Error appears on every try to open ENS settings screen. Though ENS itself is successfully added. 

[status_logs.zip](https://github.com/status-im/status-mobile/files/10812781/status_logs.zip)

photo_2023-02-23 12 58 45

https://user-images.githubusercontent.com/97245802/220887979-bc5bdd6e-13bd-42a9-b422-eeef88c4178b.mp4

**Comment by pavloburykh** [comment]
### ISSUE 4 Added nickname is not displayed in nickname section on user's profile screen

**Steps:** 

1. User A opens profile screen of User B
2. User A sets nickname for User B
3. See if nickname is displayed in nickname section 

**Expected result:** nickname is displayed

**Actual result:** "none" value is displayed despite nickname has been set

[Status-debug-logs (6).zip](https://github.com/status-im/status-mobile/files/10813141/Status-debug-logs.6.zip)

https://user-images.githubusercontent.com/97245802/220896804-24140353-fda3-4426-9b9b-380699bda336.mp4

**Comment by pavloburykh** [comment]
@flexsurfer I am still facing issues which I have described in the PR with hotfix for release branch https://github.com/status-im/status-mobile/pull/15086#issuecomment-1430229495

Are those issues in scope of current PR or they are subjects for addressing in some future PRs?

**Comment by pavloburykh** [comment]
### ISSUE 5 Regular text instead of mention is sent when trying to mention user by displayed name

Unfortunately still facing this issue which I have already reported in previous [PR](https://github.com/status-im/status-mobile/pull/15086#issuecomment-1430229370) 

It is reproducible only with some display names and there are no clear steps or preconditions on how to reproduce that. 

Reproducible for me in communities/1-1 chats/group chats

@flexsurfer you can try to reproduce the bug with the same user and steps which I have provided before. 

I have updated video and logs from current PR build. This time you can observe 2 different display names (Pavlo and Pavlo-test2), both are not resolved. 

**Preconditions:** use this user for reproduction. seed phrase: outdoor portion desk extend lend ugly canvas inquiry ski buffalo custom loud

**Steps:** 

1. Recover given above user
2. Tap Community tab
3. Wait until communities are restored
4. Enter "testing release /1.22" community
5. Open any channel in given community
6. Try to mention desktop user Pavlo or Pavlo-test2
7. Observe the result

**Expected result:** it is possible to mention user. 

**Actual result:** Regular text instead of mention is sent. Same happens in 1-1/group chats. 

[Status-debug-logs (7).zip](https://github.com/status-im/status-mobile/files/10813655/Status-debug-logs.7.zip)

https://user-images.githubusercontent.com/97245802/220912277-74881d1d-a1f9-46cb-91cb-57117d8e1056.mp4

**Comment by pavloburykh** [comment]
### ISSUE 6 Unable to remove user from contact list (Remove from contacts button is not responding)

**Preconditions:** User A and User B are mutual contacts

**Steps:** 
1. Try to remove User B from User's A contact list (or vice verse): by User A open profile screen of User B => tap Remove from contacts
2. See if Remove from contacts button responds and user is removed from contacts

**Expected result:** Remove from contacts button responds and user is removed from contacts

**Actual result:** Remove from contacts button does not respond, unable to remove user from contacts

[Status-debug-logs (8).zip](https://github.com/status-im/status-mobile/files/10814189/Status-debug-logs.8.zip)

https://user-images.githubusercontent.com/97245802/220928104-e70e191c-6e99-484d-9612-325b68dab3a1.mp4

**Comment by pavloburykh** [comment]
### ISSUE 7 3 random names are not shown under ENS/display names for contacts in contact list and profile screens. Chat keys previews for contacts with 3 random names. 

**Preconditions:** login via user who has contacts with 3 random names, ENS, display names

**Steps:** 

1. Open contact list (the old UI screen) 
2. See if chat key previews are displayed for contacts with 3 random names
3. See if 3 random names are displayed for contacts with ENS, display names
4. Open profile screens of different contacts
5. See if chat key previews are displayed for contacts with 3 random names
6. See if 3 random names are displayed for contacts with ENS, display names

**Expected result:** 3 random names and chat keys previews are shown. 

photo_2023-02-23 16 45 12


**Actual results:** 3 random names and chat keys previews are not shown. 

photo_2023-02-23 16 45 41

photo_2023-02-23 16 45 45

**Comment by flexsurfer** [comment]
hey @pavloburykh thanks, if its not like really functional i wouldn't fix it in old screens, only in new , so issues 7 looks like ok for me

**Comment by flexsurfer** [comment]
ISSUE 6 Unable to remove user from contact list (Remove from contacts button is not responding)
```
[Fri Feb 24 2023 09:27:03.325]  ERROR    ERROR [status-im2.contexts.contacts.events:86] - failed to add contact 0x04cafae3924811bad6f9aa9b05a6ce941f0903ab6dc46241c9755a853921d62e9631fd5277f0e1a828a52f63db150744bf22a2b7a2c017c2ee635f7fbc7d0c7efa {:code -32000, :message "no such table: activity_center_states"}

```
cc @cammellos

**Comment by cammellos** [comment]
@flexsurfer the issue with migration is fixed, if you update status-go, thanks!

**Comment by flexsurfer** [comment]
thanks @pavloburykh all should be fixed except mentions

**Comment by pavloburykh** [comment]
@flexsurfer thanx, will check! 
Could you please take a look why IOS build has failed? https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-15085/11/consoleText

**Comment by pavloburykh** [comment]
@flexsurfer take a look a new issue please

### ISSUE 8 Unable to start a chat with contact: chat screen is not loaded, "Add Unknown to send a message" is displayed

**Preconditions:** Users A and B are mutual contacts. No active 1-1 chat exists yet between User A and B. 

**Steps:** 

1. User A initiates 1-1 chat with User B: tap + => New chat => select User B in the list => tap Chat with User B
2. See if chat screen is loaded and User A is able to send messages

**Expected result:** chat screen is loaded and User A is able to send messages

**Actual result:** chat screen is not loaded (loading skeleton is displayed). Chat input field is disabled by "Add Unknown to send a message" label. 

[Status-debug-logs (10).zip](https://github.com/status-im/status-mobile/files/10824119/Status-debug-logs.10.zip)

photo_2023-02-24 14 35 22

https://user-images.githubusercontent.com/97245802/221180631-6e35ed49-28b3-4335-87b4-8ebc9d03cd79.mp4

**Comment by flexsurfer** [comment]
thanks @pavloburykh fixed

**Comment by pavloburykh** [comment]
### ISSUE 9 "Cannot read property 'toLowerCase' of null' error on opening Group details

**Steps:** 

1. Create a group: open Chats tab => tap + => New chat => Select at least 2 users from contact list => Setup group chat => Create group chat
2. Open Chats tab
3. Longtap on newly created group
4. Tap Group detaills
5. Observe an error

**Expected result:** no error. Group details screen is opened. 

**Actual result:** "Cannot read property 'toLowerCase' of null' error  appears

[Status-debug-logs (11).zip](https://github.com/status-im/status-mobile/files/10826921/Status-debug-logs.11.zip)

https://user-images.githubusercontent.com/97245802/221243705-2052f347-eb19-47cd-8a3a-38a82c4fed86.mp4

**Comment by flexsurfer** [comment]
thanks @pavloburykh fixed

**Comment by churik** [comment]
@flexsurfer @pavloburykh thank you for your work! I also verified PR briefly with all basic testcases with different names (ENS, display name, nickname, 3-random) - no issues from my side.

e2e failures are not related to a particular PR, I'm working today on the leftovers for fixing them.
